bloom-housing/bloom
# feat: managing preferences

# Pull Request Template

## Issue Overview

This PR addresses #2748 #2750

[Figma design](https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=5725%3A182890)


## Description

This PR implements the two issues above, which allow you to manage preferences in the new `Settings` panel in partners.
If you login as an admin or a jurisdictional admin, you should be able to see the preferences from the jurisdictions you have access to.

From the main page you can add a preference, duplicate a preference, or edit a preference.

Note some differences from the design that were decided in Slack over the last couple of weeks:
* We added the "Is this option multi-select or exclusive?" question on an individual option for preferences like `Housing Authority Project-Based Voucher - Alameda` in which there is the generic opt out value, but also a custom one (`Do not consider me`). Since `Do not consider me` is also a "pass" option, it needs to behave exclusively (it cannot be checked with any other checkbox selected). If all options are checked as exclusive, the question renders as a radio.
* We added the "Can you opt out?" question to a preference, for preferences like `Housing Authority Project-Based Voucher - Alameda` - there are two optional items you can select, but you don't have to select anything.
* We removed the `Subtitle` field visually from the form under the assumption it is not often used, but it may be added back in the future. The data is maintained, but it is hidden from the listing view, application, and preferences form.
* We removed the delete functionality for a preference over concerns around how we might handle existing listings that are tied to those preferences, and may add it back in the future.
* We added a link to the root preference in the design since it is in the existing data model.
* The data model supports multiple jurisdictions, we are starting with just one.
* The data model supports multiple links, we are starting with just one.

## How Can This Be Tested/Reviewed?

Note that the vast majority of this experience will be as a jurisdictional admin, so I'd recommend doing most of your testing there, but a super admin will still have access to all preferences across jurisdictions.

Compare visually with the [Figma design](https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=5725%3A182890) noting the differences from the design above.

Ensure the preferences populate according to your user.
Ensure you can add a preference with options, and exercise all the fields.
Ensure you can edit a preference / option.
Ensure you can duplicate a preference / option.
Ensure you can delete an option.
Ensure you can reorder options.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
